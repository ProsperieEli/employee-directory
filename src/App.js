import logo from "./logo.svg";
import "./App.css";
import Home from "./components/home/Home";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Login from "./components/login/Login";

import EmployeeList from "./components/employee list/EmployeeList";

function App() {
  return (
    <Router>
      <Switch>
        <Route exact="/">
          <Home />
        </Route>
        <Route to="/login">
          <Login />
        </Route>
        <Route to="/list">
          <EmployeeList />
        </Route>
      </Switch>
    </Router>
  );
}

export default App;
