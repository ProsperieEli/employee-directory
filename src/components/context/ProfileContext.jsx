import { useContext, createContext, useState } from "react";
import { getProfile } from "../../services/Profiles";

const profileContext = createContext();

const ProfileProvider = ({ children }) => {
  const currentProfile = getProfile();
  const [profile, setProfile] = useState(
    currentProfile ? { id: currentProfile.id, email: currentProfile.email } : {}
  );

  return (
    <profileContext.Provider value={{ profile, setProfile }}>
      {children}
    </profileContext.Provider>
  );
};

const useProfile = () => {
  const context = useContext(profileContext);

  if (context === undefined) {
    throw new Error("Incorrect placement");
  }
  return context;
};

export { ProfileProvider, profileContext, useProfile };
