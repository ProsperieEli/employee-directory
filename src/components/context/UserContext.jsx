import { useContext, createContext, useState } from "react";
import { getUser } from "../../services/Users";

const userContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  );

  return (
    <userContext.Provider value={{ user, setUser }}>
      {children}
    </userContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(userContext);

  if (context === undefined) {
    throw new Error("Incorrect placement");
  }
  return context;
};

export { UserProvider, useUser };
