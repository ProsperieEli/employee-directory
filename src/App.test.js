import { render, screen } from "@testing-library/react";
import App from "./App";
import { UserProvider } from "../src/components/context/UserContext";
import { ProfileProvider } from "../src/components/context/ProfileContext";
import { MemoryRouter } from "react-router-dom";

test("renders learn react link", () => {
  const { container } = render(
    <MemoryRouter initialEntries={["/"]}>
      <UserProvider>
        <ProfileProvider>
          <App />
        </ProfileProvider>
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
