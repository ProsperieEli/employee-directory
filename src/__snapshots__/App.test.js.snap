// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div>
    <header>
      <p>
        Not signed in
      </p>
      <a
        href="/login"
      >
        <button>
          Sign In
        </button>
      </a>
    </header>
  </div>
  <div>
    <h1>
      Welcome Employee!
    </h1>
    <form>
      <input
        id="email"
        type="email"
        value=""
      />
      <input
        id="password"
        type="password"
        value=""
      />
      <button>
        Sign up
      </button>
    </form>
    <a
      href="/login"
    >
      Already have an account? Login here!
    </a>
  </div>
</div>
`;
